examples/example_events.rs

```diff
@@ -100,23 +117,24 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
     woab::run_actix_inside_gtk_event_loop()?;
 
     woab::block_on(async {
-        factories
-            .win_app
-            .instantiate()
-            .actor()
-            .connect_signals(WindowSignal::connector().inhibit(|signal| match signal {
-                WindowSignal::AllCharactersEntryKeyPressed(_, event) => {
-                    let character = event.get_keyval().to_unicode();
-                    let is_digit = character.map(|c| c.is_digit(10)).unwrap_or(false);
-                    Some(gtk::Inhibit(is_digit))
-                }
-                _ => None,
-            }))
-            .create(|ctx| WindowActor {
-                widgets: ctx.widgets().unwrap(),
-                press_times: Default::default(),
-                total_durations: Default::default(),
-            });
+        factories.win_app.instantiate().connect_with(|bld| {
+            bld.get_object::<gtk::ApplicationWindow>("win_app").unwrap().show();
+            woab::NamespacedSignalRouter::new()
```

Comment:
In this example, where does the name space come from? I don't see any strings. I guess we route everything to everybody at the moment?

In any case, I'd like the namespace to be treated as some kind of opaque prefix, and restrict all further processing to that assumption.

Reply:
The namespace comes from the name of the actor's type. So in
```rust
.route(
    PressCountingActor {
        widgets: bld.widgets().unwrap(),
        press_times: Default::default(),
        total_durations: Default::default(),
    }
    .start(),
)
```

The method `route::<PressCountingActor>()` takes the name of the type, strips it's prefix and qualifies, and gets the namespace `"PressCountingActor"`.

If you wanted explicit namespace, you could do this:
```rust
.route_ns(
    "PressCountingActor",
    PressCountingActor {
        widgets: bld.widgets().unwrap(),
        press_times: Default::default(),
        total_durations: Default::default(),
    }
    .start()
    .recipient(),
)
```

But if we want to pass the signal name untouched, without stripping the namespace, we want to use automatic namespaces to prepare in advance for the signal handling macro.

The macro will probably look something like this:

```rust
#[woab::signal_handling]
impl PressCountingActor {
    fn press(&mut self, _: (), #[signal_handling(event)] event: gdk::EventButton) -> gtk::Inhibit {
        // ...
    }

    fn release(&mut self, _: (), #[signal_handling(event)] event: gdk::EventButton) -> gtk::Inhibit {
        // ...
    }
}
```

Which will get translated to:
```rust
impl actix::Handler<woab::Signal> for PressCountingActor {
    type Result = woab::SignalResult;

    fn handle(&mut self, msg: woab::Signal, _ctx: &mut Self::Context) -> Self::Result {
        Ok(match msg.name() {
            "PressCountingActor::press" => {
                // ...
            }
            "PressCountingActor::release" => {
                // ...
            }
            _ => msg.cant_handle()?,
        })
    }
}
```

We are using the name of the class as the namespace (this was your suggestion) so the macro knows what it is can add it to the generated code on one this side, and the router knows the type on its side because it knows the type of the actor.

Alternatively, if the namespace was to be stripped from the signal name, the macro could just generate namespace agnostic `match` branches, and you would set the namespace in the router `route()` calls to whatever you want (that matches the Glade file, of course)

Reply:
I still think using the automatic struct name is a good default, however we must account for the case where two actors have the same name in different modules. If we strip the name from the event handling, this makes the macro care less, which I like. How about providing both `msg.name()` and `msg.full_name()` (suggestion open to bike shedding) on the signal, so the choice is up to the handler?

The macro looks exactly how I'm dreaming it to be :D. But if this is too complex, I also have an idea for a non-proc macro that replaces and enhanced the `match` statement in the signal handler, taking care of getting the arguments.

Reply:
Wait! I know!

The router will have three methods:

|                                      | **Strip Namespace** | **Keep Namespace** |
| ------------------------------------ | ------------------- | ------------------ |
| **Manually Specified Namespace**     | `route_strip_ns()`  | `route_ns()`       |
| **Automatically Detected Namespace** | `route()`           |                    |

The rationale is that if you are using the automatic namespace detection, you can safely strip it because it's encoded by the type name. If you are not using it, and write your own namespace, you won't be able to grep for it that easily - but if we keep it in the signal string you can grep for it and find the `match` arm. And if for whatever reason you need to manually specify it but still strip it you can use the one with the longest name - `route_strip_ns()`.